src/Microsoft.OData.Core/UriParser/Visitors/PathSegmentToContextUrlPathTranslator.cs

```diff
@@ -180,6 +180,36 @@ public override string Translate(CountSegment segment)
             return string.Empty;
         }
 
+        /// <summary>
+        /// Translate a FilterSegment
+        /// </summary>
+        /// <param name="segment">the segment to Translate</param>
+        /// <returns>Defined by the implementer.</returns>
+        public override string Translate(FilterSegment segment)
+        {
+            return "/" + segment.FullSegment;
+        }
+
+        /// <summary>
+        /// Translate a ReferenceSegment
+        /// </summary>
+        /// <param name="segment">the segment to Translate</param>
+        /// <returns>Defined by the implementer.</returns>
+        public override string Translate(ReferenceSegment segment)
+        {
+            return string.Empty;

```

Comment:
This actually came up in our TC meeting today.  I'm unclear what we should do in a ContextUrl for a reference segment. Please open an issue to track, and reference it from the code here.

Reply:
Adding https://github.com/OData/odata.net/issues/1292 to code comments there.